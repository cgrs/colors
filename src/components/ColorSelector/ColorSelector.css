.color-selector {
  flex-grow: 1;
}
.color-selector.animated {
  animation: grow 200ms ease 0s 1 forwards;
}
@keyframes grow {
  0% {
    flex-grow: 0;
  }
  100% {
    flex-grow: 1;
  }
}
