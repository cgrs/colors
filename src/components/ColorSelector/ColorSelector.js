import React from 'react'
import './ColorSelector.css'

const ColorSelector = ({color, reference, ...props}) => (
  <div
    ref={reference}
    style={{
      backgroundColor: color.string()
    }}
    {...props}></div>
)

export default ColorSelector
