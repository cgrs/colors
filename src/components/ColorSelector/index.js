import ColorSelector from './ColorSelector'
export default ColorSelector
