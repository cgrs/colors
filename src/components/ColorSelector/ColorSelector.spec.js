import React from 'react'
import ColorSelector from './ColorSelector'
import renderer from 'react-test-renderer'

describe('ColorSelector', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ColorSelector color={{
      string: () => '#000000'
    }}/>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
