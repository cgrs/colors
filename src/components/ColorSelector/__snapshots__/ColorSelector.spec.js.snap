// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#000000",
    }
  }
/>
`;
